Hex space social graph milestone 1

rust-0x0 opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request:https://github.com/w3f/Grants-Program/pull/1020

takahser commented:
Thanks for the delivery, [rust-0x0](https://github.com/rust-0x0)! I'll look into it shortly.

takahser reviewed (changes requested):
@rust-0x0 you can find my evaluation [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/hex-space-social-graph_1_takahser.md). There are some small things that should be improved or clarified. Most importantly, please add some instructions on how to test the frontend.

Also, just FYI (this is not relevant for the acceptance of the delivery) - some of the social links at the end of your [article](https://medium.com/@rust.0x0/hex-space-social-graph-487c548c2391) are broken.

rust-0x0 commented:
> here

Thanks for the review.I will update it  as soon as possible.

rust-0x0 commented:
what is the url link
of the node based on the docker image ? 

front-end config guide

https://github.com/rust-0x0/hex-space-protocol-docs/tree/milestone-1#setup-hexspacesocialgraph-protocol-front-end


https://github.com/rust-0x0/hex-space-protocol-front-end/tree/362c896bb11f91fbe0c1728b93b28ab6db4963d0#setup-hexspacesocialgraph-protocol-front-end

front-end tutorial about follow &profile
https://github.com/rust-0x0/hex-space-protocol-docs/blob/milestone-1/Follow.md
https://github.com/rust-0x0/hex-space-protocol-docs/blob/milestone-1/Profile.md


note 
prerequisite
.env  
REACT_APP_PROVIDER_SOCKET=ws://127.0.0.1:9944
REACT_APP_CONTRACT_ADDRESS=5H4rkHhc6w1A95GDMDuFoTQ6MZcjxY4N5aHiUrSncXDrSasR

update correct  contract address


it will show dev account list  if success 
it will show 'No followings to be shown' if failed

takahser reviewed (changes requested):
@rust-0x0 thanks for your reply.
I followed your follow and profile guides. My polkadot accounts are displayed properly, I'm also able to switch back-and-forth between them. However, when trying to send the transactions by clicking "Batch Follow" and "Save on chain", in both cases there are console errors and it seems like the transactions don't go through:

<img width="713" alt="image" src="https://user-images.githubusercontent.com/5393704/179986042-dab0ad19-f1c2-45f2-9003-2b14cb7ae772.png">

Note that I'm still testing on the rococo contracts parachain.

rust-0x0 commented:
if start front end from source code
need update  provider rpc url  and   manul deployed ccontrat address  to .env file
rename example.env to .env 
> .env
> REACT_APP_PROVIDER_SOCKET=ws://127.0.0.1:9944
> REACT_APP_CONTRACT_ADDRESS=5H4rkHhc6w1A95GDMDuFoTQ6MZcjxY4N5aHiUrSncXDrSasR
if  Start front-end  using docker
# update the correct  contract address to env REACT_APP_CONTRACT_ADDRESS
docker run -e REACT_APP_PROVIDER_SOCKET=ws://127.0.0.1:9944  -e REACT_APP_CONTRACT_ADDRESS=5H4rkHhc6w1A95GDMDuFoTQ6MZcjxY4N5aHiUrSncXDrSasR -p 8100:8100  rust0x0/hex-space-protocol-front-end:0.1

## Start front-end  using docker-compose
update env REACT_APP_CONTRACT_ADDRESS  to the correct  contract address  in docker-compose.yml file 
```bash
 environment:
      - REACT_APP_PROVIDER_SOCKET=ws://127.0.0.1:9944
      - REACT_APP_CONTRACT_ADDRESS=5H4rkHhc6w1A95GDMDuFoTQ6MZcjxY4N5aHiUrSncXDrSasR
```

rust-0x0 commented:
- highlight  front end test guide
- update link with commit hash in delivery doc
- fixed dockfile biuld issue
- fixed front end issues
- add https://rust-0x0.github.io/hex-space-protocol-front-end   support rococo contract parachain   (stop producing block by now)  and simply test front end
please review again at your convenience

takahser commented:
@rust-0x0 what's the reason you're using the [contracts-node in v0.3.0](https://github.com/rust-0x0/hex-space-protocol-docs/tree/milestone-1#21-1-installing-the-substrate-contracts-node)? I'm not sure why, but when I ran the contract node again, the "Contracts" entry was missing in the navigation.

> add https://rust-0x0.github.io/hex-space-protocol-front-end support rococo contract parachain (stop producing block by now) and simply test front end

Does that indicate that I should now use Rococo for testing your delivery instead?

rust-0x0 commented:
> @rust-0x0 what's the reason you're using the [contracts-node in v0.3.0](https://github.com/rust-0x0/hex-space-protocol-docs/tree/milestone-1#21-1-installing-the-substrate-contracts-node)? I'm not sure why, but when I ran the contract node again, the "Contracts" entry was missing in the navigation.
> 
> > add https://rust-0x0.github.io/hex-space-protocol-front-end support rococo contract parachain (stop producing block by now) and simply test front end
> 
> Does that indicate that I should now use Rococo for testing your delivery instead?

Sorry, [contracts-node in v0.3.0] is the mistake of the doc. I will update the doc.
it supports  the latest  version of contracts-node  .
you   could use the latest  version of contracts-node 
or  Rococo for testing my delivery.

takahser commented:
@rust-0x0 I tested on Rococo for now (I changed the ENV variables accordingly) according to your [guide](https://github.com/rust-0x0/hex-space-protocol-docs/blob/849c659bb76c1029ce17b7c5f7c19c2e3bb070d0/Follow.md).

The follow feature works, however, the other features still didn't work for me:

- The unfollow feature doesn't work at all: Neither when separately unfollowing, nor when batch unfollowing. There are no console errors. Also, reloading the page doesn't change the state neither.
- Same for the contract approval feature: Nothing happens.
- The `TransferFrom` feature should transfer NFTs, right? Currently, the 3rd dropdown is showing me a list of my accounts, rather than a list of NFTs.
- Same goes for `BatchTransferFrom`.

rust-0x0 commented:
The unfollow feature doesn't work at all: Neither when separately unfollowing, nor when batch unfollowing. There are no console errors. Also, reloading the page doesn't change the state neither.
The current account approval  hex contract  before  unfollow .
Same for the contract approval feature: Nothing happens.
 After The current account approval  hex contract ,    unfollow   throughing hex contract call  ERC-1155 contract 's burn function .
The TransferFrom feature should transfer NFTs, right? Currently, the 3rd dropdown is showing me a list of my accounts, rather than a list of NFTs.
Same goes for BatchTransferFrom.
The  logic of TransferFrom  or BatchTransferFrom  :
Alice transfer Bob (inside code convert to tokenId) to charlie  Before  Charlie followed Alice   and Alice followed Bob.

rust-0x0 commented:
There are   gas estimation issues of some features  when I'm testing on Rococo .I'm solving these issue.

takahser commented:
@rust-0x0 thanks, I'll wait for your update. Do you have any idea on why it works on the local blockchain, but not on the rococo contracts parachain?

rust-0x0 commented:
> 
The possible  of   the version of polkadot-js  in the hex space front end .
Because in the polkadot-apps contracts ,there is  no issue  when  calling `transferform` or `set_info`  method.
  
I'm testing and position the issue. I will solve the issue asap.

rust-0x0 commented:
1. fixed gas estimation issue in the some features on rococo.
2. fixed the version issue of the readme in the docs.
please review again at your convenience

takahser commented:
@rust-0x0 thanks for your patience and sorry for the wait.
I tested again, here are my findings:

- _Unfollow_ works now, but with plenty of console errors:
  <img width="498" alt="image" src="https://user-images.githubusercontent.com/5393704/185384545-ccd12d00-5a87-4519-95cb-23d72c6ef948.png">
- The same goes for the _Batch Unfollow_ feature:
  <img width="479" alt="image" src="https://user-images.githubusercontent.com/5393704/185385395-8b8ea083-9c15-4c38-8617-e31da9daa34c.png">
- _Batch Follow_ works, lots of console errors here as well:
  <img width="492" alt="image" src="https://user-images.githubusercontent.com/5393704/185385816-55b8f228-82bb-4411-a1f6-76b674d783ab.png">
- _Approval_ works:
  <img width="356" alt="image" src="https://user-images.githubusercontent.com/5393704/185385529-8386bbc4-0081-459e-8f7e-1ce40006eda9.png">

However, the _TransferFrom_ and _BatchTransferFrom_ features don't work:

<img width="485" alt="image" src="https://user-images.githubusercontent.com/5393704/185390189-3b80aff1-758b-4755-ac33-5671b82a20ba.png">

Also, I still don't understand what exactly is supposed to be transferred here and how I can look up the balances of these assets (NFTs I guess). I didn't quite get, what you [meant](https://github.com/w3f/Grant-Milestone-Delivery/pull/500#issuecomment-1213098657) by:

> Alice transfer Bob (inside code convert to tokenId) to charlie Before Charlie followed Alice and Alice followed Bob.

rust-0x0 commented:
>   the caller's  Roc Balance should be too low.
TransferFrom and BatchTransferFrom features  test steps:

1. Alice   follows  Bob (    mint  the Token Id(Bob's address generated) to Alice).
2. Charlie follows Alice( Charlie becomed the follower of Alice).  Charlie is not the follower of Bob.
3. the caller is Alice  (the current account is Alice).
4. The Transferfrom 's parameter   `from ` is Alice  ,`to` is Charlie   , `TokenId ` is Bob.
5. The Execute  result  is :  The TokenId (Bob )  is transfered from Alice to Charlie; Charlie is the follower of Bob;Alice is not the follower of Bob.

There are  some ` CONTRACT: Unable to decode contract event: Unable to find event with index` issues.
When using  the latest  cargo-contract build hexspace contract .
When  the extrinsic gas estimation failed   (print `result is err `) on Rococo   ,set  gas limit to -1 , the extrinsic  tx executed success.

I will handle these issues.

rust-0x0 commented:
1. fixed  the `gas estimation fail, print result.err  0x19000000` issue.
2. fixed  the `unable to  decode  contract event:Unable to find event with index ` issue.

please review again at your convenience

takahser reviewed (changes requested):
@rust-0x0 does the frontend still work properly on your machine? In my case, the table with the followers doesn't load anymore on commit 508ca3ccf0459e975445eded2b084efbc34c6a6a.

<img width="953" alt="image" src="https://user-images.githubusercontent.com/5393704/187202281-3e5148bf-72cf-4fd3-b909-b546c82b5a38.png">

This used to work in commit ff8b9a2e7e4a34ba25e9f94b4c90abda2112c0a3.

rust-0x0 commented:
> https://rust-0x0.github.io/hex-space-protocol-front-end

update   new deployed contract   address to  the env   the front end


https://rust-0x0.github.io/hex-space-protocol-front-end/
is  the latest  front end  with new deployed contract .

rust-0x0 commented:
I pull the code of the commit 508ca3ccf0459e975445eded2b084efbc34c6a6a.
I run and test it on Rococo.
There is no issue.

takahser commented:
@rust-0x0

> I run and test it on Rococo.
> There is no issue

That's mysterious, since it doesn't work for me on 508ca3ccf0459e975445eded2b084efbc34c6a6a anymore. Does it work for you with ff8b9a2e7e4a34ba25e9f94b4c90abda2112c0a3 as well, using the same contract? Did you update the contract recently? I didn't see any updates.

rust-0x0 commented:
No, It is not  .It is new contract name(hex_space_social_graph),new folder.
Sorry,  I updated the code link .but I don't update contract name. ` ink! Contract: HexSpace` in `deliveries/hex-space-social-graph_milestone_1.md`.
It is  `ink! Contract: HexSpaceSocialGraph`. it  willl more highlight or noticeable.


```
 1. |  ink! Contract: HexSpace | [code](https://github.com/rust-0x0/hex-space-protocol-substrate/tree/9123f9503f45a89958bef6c96eb308b82dd7f3a6/hex_space_social_graph) |  |
 ```
 [In https://github.com/rust-0x0/Grant-Milestone-Delivery/commit/d1722091595a04c99c036da08696991305cbc413](https://github.com/rust-0x0/Grant-Milestone-Delivery/blob/d1722091595a04c99c036da08696991305cbc413/deliveries/hex-space-social-graph_milestone_1.md)
  
 the code  link is the new contract  name (hex_space_social_graph)  .
 
 


 

https://github.com/rust-0x0/hex-space-protocol-substrate/tree/9123f9503f45a89958bef6c96eb308b82dd7f3a6/hex_space_social_graph     in   

>

rust-0x0 commented:
Thanks for your review .please review again at your convenience.

takahser reviewed (approved):
@rust-0x0 so you created a new contract in [hex_space_social_graph](https://github.com/rust-0x0/hex-space-protocol-substrate/tree/9123f9503f45a89958bef6c96eb308b82dd7f3a6/hex_space_social_graph) instead of updating the existing contract at [hex_space](https://github.com/rust-0x0/hex-space-protocol-substrate/tree/9123f9503f45a89958bef6c96eb308b82dd7f3a6/hex_space), is that correct? Is there any particular reason why you didn't just update the existing one or at least remove the legacy contract since it's not relevant anymore?

That being said, I took another look and with the new contract the frontend works fine. I successfully tested all the methods using the frontends, also transfer and batchTransfer finally work now.
I hereby accepted your delivery, you can find my evaluation [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/f7dee284f7002f25f31c6759f42fa076adc089da/evaluations/hex-space-social-graph_1_takahser.md).

rust-0x0 commented:
> 
Right,no particular reason.I will  remove the legacy contract.It 's not relevant anymore.
At that time ,Keep it temporarily, in order to compare the old and new contracts.
Thank you.

takahser commented:
@rust-0x0 ok sounds good. Also, feel free to merge the milestone 1 branch into your main branch.
I was about to forward your invoice, but I wasn't able to find it. Did you submit it, and if so, do you remember the date or filename?

rust-0x0 commented:
> 
I submitted my invoice  on July 13th  .  the filename is `USDTInvoice.pdf`.
I merged  the contract and front end code ,docs into main branch.

RouvenP commented:
hi @rust-0x0 we transferred the payment today.

rust-0x0 commented:
I have received the payment .Thank you.

On Fri, Sep 9, 2022 at 10:53 PM RouvenP ***@***.***> wrote:

> hi @rust-0x0 <https://github.com/rust-0x0> we transferred the payment
> today.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grant-Milestone-Delivery/pull/500#issuecomment-1242078024>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVPUJTVPUA5ZXZ77CTBK3OTV5NFO7ANCNFSM53RECHBQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
